Framework/MDAlgorithms/src/ConvertHFIRSCDtoMDE.cpp

```diff
@@ -0,0 +1,308 @@
+          inputWSmsg << "Log " << log << " has incorrect lenght, ";
+      } else {
+        inputWSmsg << "Missing required log " << log << ", ";
+      }
+    }
+  }
+  if (!inputWSmsg.str().empty())
+    result["InputWorkspace"] = inputWSmsg.str();
+
+  std::vector<double> minVals = this->getProperty("MinValues");
+  std::vector<double> maxVals = this->getProperty("MaxValues");
+
+  if (minVals.size() != 3 || maxVals.size() != 3) {
+    std::stringstream msg;
+    msg << "Must provide 3 values, 1 for every dimension";
+    result["MinValues"] = msg.str();
+    result["MaxValues"] = msg.str();
+  } else {
+    std::stringstream msg;
+
+    size_t rank = minVals.size();
+    for (size_t i = 0; i < rank; ++i) {
+      if (minVals[i] >= maxVals[i]) {
+        if (msg.str().empty())
+          msg << "max not bigger than min ";
+        else
+          msg << ", ";
+        msg << "at index=" << (i + 1) << " (" << minVals[i]
+            << ">=" << maxVals[i] << ")";
+      }
+    }
+
+    if (!msg.str().empty()) {
+      result["MinValues"] = msg.str();
+      result["MaxValues"] = msg.str();
+    }
+  }
+
+  return result;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void ConvertHFIRSCDtoMDE::init() {
+
+  declareProperty(std::make_unique<WorkspaceProperty<API::IMDHistoWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "An input workspace.");
+  declareProperty(
+      std::make_unique<PropertyWithValue<double>>(
+          "Wavelength", DBL_MAX,
+          boost::make_shared<BoundedValidator<double>>(0.0, 100.0, true),
+          Direction::Input),
+      "Wavelength");
+  declareProperty(
+      std::make_unique<ArrayProperty<double>>("MinValues", "-10,-10,-10"),
+      "It has to be 3 comma separated values, one for each dimension in "
+      "q_sample."
+      "Values smaller then specified here will not be added to "
+      "workspace.");
+  declareProperty(
+      std::make_unique<ArrayProperty<double>>("MaxValues", "10,10,10"),
+      "A list of the same size and the same units as MinValues "
+      "list. Values higher or equal to the specified by "
+      "this list will be ignored");
+  // Box controller properties. These are the defaults
+  this->initBoxControllerProps("5" /*SplitInto*/, 1000 /*SplitThreshold*/,
+                               20 /*MaxRecursionDepth*/);
+  declareProperty(std::make_unique<WorkspaceProperty<API::IMDEventWorkspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "An output workspace.");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void ConvertHFIRSCDtoMDE::exec() {
+  double wavelength = this->getProperty("Wavelength");
+  if (wavelength == Mantid::EMPTY_DBL()) {
```

Comment:
Isn't this taken care of by the bounded validator in `init`?

Reply:
Yes, that was there from before I added it to the validation so I can remove it.